File: Assets/MoveAround.js
#pragma strict

var planets;

function Start () {
	if (planets == null) {
		planets	= GameObject.FindGameObjectsWithTag ("Planet");
	}
}

var speed = 10.0;
var rotateSpeed = 10.0;

function Update () {
	var controller : CharacterController = GetComponent(CharacterController);
	
	transform.Rotate(0, Input.GetAxis("Horizontal") * rotateSpeed, 0);
	
	var forward = transform.TransformDirection(Vector3.forward);
	var curSpeed = speed * Input.GetAxis("Vertical");
	controller.SimpleMove(forward * curSpeed);

	var position = transform.position; 
	for (var planet : GameObject in planets) {
		var diff = (planet.transform.position - position);
		var curDistance = diff.sqrMagnitude; 
		Debug.Log(curDistance);
	}

}
@script RequireComponent(CharacterController)

File: Assets/MoveAround.cs
using UnityEngine;
using System.Collections;

public class MoveAround : MonoBehaviour
{	
	float rotateForce = 1f;
	float moveForce = 3f;
	
	void Start ()
	{
	}
	
	void Update ()
	{
		float hInput = Input.GetAxis ("Horizontal");
		float vInput = Input.GetAxis ("Vertical");
		
		if (hInput > 0 || vInput > 0) 
		{
			Debug.Log (string.Format ("hInput: {0}\nvInput: {1}", hInput, vInput));
		}
		
		gameObject.rigidbody.AddTorque (gameObject.transform.up * rotateForce * hInput, ForceMode.Force);
		gameObject.rigidbody.AddForce (gameObject.transform.forward * moveForce * vInput, ForceMode.Force);
	}
}
